dom.go

```diff
@@ -13,8 +13,8 @@ var batch = &batchRenderer{idx: make(map[Component]int)}
 // core/central struct which all Component implementations should embed.
 type Core struct {
 	prevRenderComponent Component
-	prevRender          *HTML
-	unmounted           bool
+	prevRender          ComponentOrHTML
+	mounted, unmounted  bool
```

Comment:
I don't follow why we now have two booleans for this.

Would we end up in situations where `unmounted == true && mounted == true` ? The fact that we have two booleans seems to imply that.

Why not just keep `mounted` and use `!mounted` to indicate `unmounted` ?

Reply:
The zero value makes this difficult, and users may not implement both Mounter and Unmounter interfaces.  I'll take a look and see if this is possible, but I have doubts.

Reply:
Had a couple of minutes to think about this, and I don't think it can be handled by a single var.  For a Component that implements `Unmounter`, but not `Mounter`:

1. Component instantiated: `Core.mounted == false` (zero value)
2. Component rendered: `Core.mounted == false` (no change)
3. Component unmounted: `Core.mounted == false` (unmount erroneously skipped)

And it's perfectly valid for a component to implement one but not the other.